### PR TITLE
fix: upload component ajax error

### PR DESCRIPTION
**Pull request by pzzyf**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by pzzyf** [comment]
cc @btea @FrontEndDog

**Review by FrontEndDog — APPROVED** [review]
LGTM!

File: packages/components/upload/src/ajax.ts [hunk 1]
```diff
@@ -70,8 +70,13 @@ export const ajaxUpload: UploadRequestHandler = (option) => {
   const formData = new FormData()
   if (option.data) {
     for (const [key, value] of Object.entries(option.data)) {
-      if (isArray(value) && value.length) formData.append(key, ...value)
-      else formData.append(key, value)
+      if (isArray(value) && value.length) {
+        for (const val of value) {
+          formData.append(key, val)
+        }
```
Comment (btea) [inline packages/components/upload/src/ajax.ts:76]: The value type of `data` has no restrictions. If the value of a certain attribute is special, then this modification may cause the original behavior that is inconsistent.
```ts
// for example
const blob = new Blob(['text'], { type: 'text/plain' })
const data = {
  test: [blob, 'testname']
}
// origin 
formData.append('test', ...[data.test])

// changed
formData.append('test', data.test[0])
formData.append('test', data.test[1])
```
The above two writing effects seem to be inconsistent. 🤔

Comment (pzzyf) [reply packages/components/upload/src/ajax.ts:76]: you are right, value type is string | Blob, formData object value type is string or Blob
image
image

File: packages/components/upload/src/ajax.ts [hunk 2]
```diff
@@ -70,10 +70,11 @@ export const ajaxUpload: UploadRequestHandler = (option) => {
   const formData = new FormData()
   if (option.data) {
     for (const [key, value] of Object.entries(option.data)) {
-      if (isArray(value) && value.length) formData.append(key, ...value)
+      if (isArray(value) && value.length) formData.append(key, value.join(','))
```
Comment (btea) [inline packages/components/upload/src/ajax.ts:73]: It seems that the stitching of multiple values ​​into one value also changes the original logic.

**Pull request closed without merging**